internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -2211,6 +2217,12 @@ Segment getSegment(int partition, boolean sendEndOfSegment) {
           long currentSegmentStartTime = segmentsStartTimeArray[partition];
           if (currentSegmentStartTime != -1
               && LatencyUtils.getElapsedTimeFromMsToMs(currentSegmentStartTime) > maxElapsedTimeForSegmentInMs) {
+            String msg = "Segment " + currentSegment + " has been open for more than" + maxElapsedTimeForSegmentInMs
+                + " ms, ending it and starting a new one.";
+
+            if (!REDUNDANT_LOGGING_FILTER.isRedundantException(msg)) {
```

Comment:
IIUC, every time this code block runs, the msg is different, so the logging throttling won't help.

Reply:
I see, I thought we will have the case where multiple threads will try to end one segment at the same time. Probably I over-complicated it. Removing it now.

Reply:
It could be, but this code block is synchronized.